xerparser/validation.py

```
@@ -0,0 +1,713 @@
+    # Check for orphan activities (no WBS)
+    if hasattr(reader, 'activities'):
+        wbs_ids = set()
+        if hasattr(reader, 'wbss'):
+            wbs_ids = {w.wbs_id for w in reader.wbss if hasattr(w, 'wbs_id')}
+
+        for activity in reader.activities:
+            if hasattr(activity, 'wbs_id') and activity.wbs_id not in wbs_ids:
+                result.add_issue(ValidationIssue(
+                    severity=ValidationSeverity.WARNING,
+                    code="XER_010",
+                    message=f"Activity references non-existent WBS",
```

Ruff (0.14.13)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)